### ISSUE TITLE
Failed to convert DB from sqlite to MariaDB

### ISSUE THREAD

**Issue by nerux1** [nextcloud/server#5099]
Hi,

I will want migrate from SQLite to MariaDB, but I have a problem when I use the `occ` command.
It seems to have a problem with the primary key of the _oc_preferences_ table.

Thanks,

### Steps to reproduce
1. `php occ db:convert-type --clear-schema mysql nextclouduser 127.0.0.1 nextcloud`
2. wait...

### Expected behaviour
Migrate from SQLite to MariaDB.

### Actual behaviour
The script exit with error.
```
[...]
    0 [>---------------------------]oc_preferences
 28/59 [=============>--------------]  47%
                                                                                                                                                                                         
  [Doctrine\DBAL\Exception\UniqueConstraintViolationException]                                                                                                                           
  An exception occurred while executing 'INSERT INTO `oc_preferences` (`userid`, `appid`, `configkey`, `configvalue`) VALUES(?, ?, ?, ?)' with params ["XXXX", "calendar", "currentView", "month"]:                                                                                                                                                                        
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'XXXX-calendar-currentView' for key 'PRIMARY'                                                                                             
  [Doctrine\DBAL\Driver\PDOException]                                                                                     
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'XXXX-calendar-currentView' for key 'PRIMARY'                                                                                             
  [PDOException]                                                                                                          
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'XXXX-calendar-currentView' for key 'PRIMARY' 
```
### Server configuration

**Operating system**: arch

**Web server:** nginx 1.12.0

**Database:** mariadb 10.1.23

**PHP version:** 7.1.5

**Nextcloud version:** 12.0.0.19

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:** package arch

**Comment by koppd** [comment]
I had the same issue. I guess your sqlite database has these 2 entries in the table oc_preferences:
- XXXX-calendar-currentView
- XXXX-calendar-currentview

I looks like that sqlite tables ARE case-sensitive and mysql table are NOT. That's why you get an error (duplicate entry for PRIMARY KEY).

I solved that issue by removing one of the two lines (I guess it was "currentview") in the sqlite database (=data/owncloud.db).
For the removal I used the "SQLite Manager" which is a Firefox Extension. 
But I'm sure there's a 2-3 liner at the terminal level, which is doing the same, but more elegant.
